website/src/react-native/versions.js

```diff
@@ -14,59 +14,114 @@ var Metadata = require('Metadata');
         path: isLatest ? '/react-native' : '/react-native/releases/' + version,
-        release: 'https://github.com/facebook/react-native/releases/tag/v' + version + '.0' + (isRC ? '-rc.0' : '')
+        release: 'https://github.com/facebook/react-native/releases/tag/v' + version + '.0' + (isRC ? '-rc.0' : ''),
+        type: isLatest ? 'latest' : (isRC ? 'release-candidate' : 'release'),
       }
     }));
 
-    if (!Metadata.config.RN_LATEST_VERSION) {
+    if (!latestVersion) {
       versions = [
         {
           title: 'current',
           path: '/react-native',
-          release: null
+          release: null,
+          type: 'latest',
         },
       ].concat(versions);
     }
 
+    var latests = versions.filter(function(version) {
+      return version.type === 'latest';
+    });
+    var masters = versions.filter(function(version) {
+      return version.type === 'master';
+    });
+    var releaseCandidates = versions.filter(function(version) {
+      return version.type === 'release-candidate';
+    });
+    var releases = versions.filter(function(version) {
+      return version.type === 'release';
+    });
+
     return (
-      <Site section="versions" title="Documentation archive">
+      <Site section="versions" title="React Native Versions">
         <section className="content wrap documentationContent nosidebar">
           <div className="inner-content">
             <h1>React Native Versions</h1>
-            <p>React Native is following a 2-week train release. Every two weeks, a Release Candidate (rc) branch is created off of master and the previous rc branch is being officially released.</p>
+            <p>React Native follows a 2-week release train. Every two weeks, a new branch created off master enters the <a href="versions.html#rc">Release Candidate</a> phase, and the previous Release Candidate branch is released and considered <a href="versions.html#latest">stable</a>.</p>
+            <a name="latest" />
+            <h3>Current Version</h3>
+            <table className="versions">
+              <tbody>
+                {latests.map((version) =>
+                  <tr>
+                    <th>{version.title}</th>
+                    <td><a href={version.path}>Documentation</a></td>
+                    <td>{version.release && <a href={version.release}>Release Notes</a>}</td>
+                  </tr>
+                )}
+              </tbody>
+            </table>
+            <p>This is the version that is configured automatically when you run <code>react-native init</code>. We highly recommend using the current version of React Native when starting a new project.</p>
+            <p>If you have an existing project that uses React Native, read the release notes to learn about new features and fixes. You can follow <a href="/react-native/docs/upgrading.html">our guide to upgrade your app</a> to the latest version.</p>
+            <a name="rc" />
+            <h3>Pre-release Versions</h3>
```

Comment:
Actually since this section contains both master and at least one RC, I don't think changing the title to `Pre-release Version (RC)` would work unless we move master somewhere else.

The RC release already has `-RC` appended to its title, would that work?